bin/oneoff/aichat_logs_to_csv

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env ruby
+
+require 'csv'
+require_relative '../../dashboard/config/environment'
+
+def main
+  options = {}
+  OptionParser.new do |opts|
+    opts.on('--section_id SECTION_ID') do |section_id|
+      options[:section_id] = Integer(section_id)
+    end
+    opts.on('--date DATE') do |date|
+      options[:date] = Date.parse(date)
+    end
+  end.parse!
+
+  CSV.open("aichat_logs_#{options[:section_id]}_#{options[:date]}.csv", "w") do |csv|
+    headers = %w[
+      session_id user_id username name level_id script_id project_id created_at
+      updated_at selectedModelId temperature systemPrompt retrievalContexts
+      status role content
+    ]
+    csv << headers
+
+    student_users_in_section = Section.find(options[:section_id]).followers.map(&:student_user)
+    sessions = AichatSession.where(
+      user: student_users_in_section,
+      created_at: options[:date].beginning_of_day..options[:date].end_of_day
+    )
+
+    sessions.each do |session|
+      model_customizations = JSON.parse(session.model_customizations)
+      messages = JSON.parse(session.messages)
+      user = session.user
+
```

Comment:
I think maybe the script level path may be more useful to the teacher than the level_id and script_id. You could provide the level path:
```
level = Level.find(session.level_id)
script_level = level.script_levels.find_by_script_id(session.script_id)
path = script_level.path
```

Reply:
Good call, updated!

Reply:
Just calling out that this may not work for sublevels (since I've been deep in similar code for bubble choice levels in lab2 😅). Sublevels I believe will not have a `script_levels` field, but if we have access to the sublevel position, we can compute this path with `build_script_level_path(script_level, {sublevel_position: index + 1})`

Reply:
haha I was wondering this as well :) Kind of expecting something might go wrong running this and might require a bit of manual editing, good to know!